ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/SandboxIndexAndWriteService.scala

```diff
@@ -353,7 +353,8 @@ abstract class LedgerBackedIndexService(
               Checkpoint(
                 domain.LedgerOffset.Absolute(Ref.LedgerString.assertFromString(offset.toString)),
                 c.recordedAt)
-            case (offset, r: LedgerEntry.Rejection) =>
+            case (offset, r: LedgerEntry.Rejection)
+                if r.commandId.nonEmpty && r.applicationId.contains(applicationId.unwrap) =>
```

Comment:
How is this case handled if the guard doesn't hold?

Reply:
With guard you mean the if-condition, right?

CommandRejected events are dropped from the ledger if commandId is not specified (i.e. a non-submitter event) or if the applicationId does not match.

Reply:
Ah, sure, thanks, it was quite hard to figure out the meaning of this line by itself, I had a better look at the context and it makes sense, thanks.